### PR TITLE
Added gl ammo for gl weapons

### LINKED ISSUE #238: Launched grenades and flares not added to shop magazines
### Expected behaviour
If you add a gl weapon and e.g. GL-Flares to the shop, the should show up when you press magazines.

### Actual behaviour
They don't, as they don't seems to be included to the magazines for gl weapons

### Steps to reproduce the behaviour 
1. It wasn't me o_O

### [RPT](https://community.bistudio.com/wiki/Crash_Files#Arma_3) and/or extDB2 logs (if applicable)
NO RPT

----

Mission version: ALL

Launcher version: ALL

Game version: CURRENT

Branch: Arma 3

### PR DESCRIPTION
**Pull request by Blauschimmel**
<Please review the guidelines for contributing to this repository. The link is above.>

Resolves #238 

#### Changes proposed in this pull request: 
- added arrays for 1rnd and 3rnd gl ammo and 1rnd gl weapons and 3rnd gl weapons
- added the ammo arrays to the respective magazine_array

### CODE REVIEW

**Comment by WolfSkin1337** [comment]
When you make this change. Then you can Add a gl weapon with smoke Grenade shells for Police. 
Because there can use it dir raid houses or something. 
Or Not?  @Jawshy @BoGuu

**Comment by Jawshy** [comment]
@WolfSkin1337, that sounds like something that should be left up to individual server owners to add.

**Comment by WolfSkin1337** [comment]
Yes you are right. Then we dont need to add a extra gl weapon with mags.
And leave it to the individual server owners to add

**Comment by Blauschimmel** [comment]
@WolfSkin1337 so you're not convinced that people should have the ability to add GL Weapons by simply adding them to a shop?

**Comment by WolfSkin1337** [comment]
All good. The change are good for to individual server owners to add the gl weapons simply

**Comment by Blauschimmel** [comment]
Ah now I get what you meant!

**Comment by danielstuart14** [comment]
that shouldn't be done that way.
You must check which GL the weapon supports using its arma config, then, compare with the ones added to the shop.

**Comment by danielstuart14** [comment]
```
_muzzles =  FETCH_CONFIG2(getArray,"CfgWeapons",_weapon,"muzzles");
if ("EGLM" in _muzzles) then {
    _eglArray = FETCH_CONFIG(getArray,"CfgWeapons",_weapon,"EGLM","magazines");
    magArray append _eglArray;
};
```

**Comment by Blauschimmel** [comment]
That's what I've been looking for. I will add this! Did you change _magArray to magArray on purpose?

**Comment by Blauschimmel** [comment]
You forgot to care about the 3GL weapons. I just added them as well, but I'm not sure about the "GL_3GL_F" in FETCH CONFIG, as the calss name is GL_3GL_F but the displayname is just "3GL". Wich one has to be used?

```
     if ("GL_3GL_F" in _muzzles) then {
        _eglArray = FETCH_CONFIG(getArray,"CfgWeapons",_weapon,"GL_3GL_F","magazines");
        _magArray append _eglArray;
    };
```

**Comment by Blauschimmel** [comment]
Works fine the way it is!

**Comment by danielstuart14** [comment]
@Blauschimmel 
Using EDEN you can see all the Configs.

Just put a player in the map, then, start the mission.
Press ESC and select CONFIG in debug monitor.

File: Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf [hunk 1]
```diff
@@ -8,11 +8,20 @@
 */
 disableSerialization;
 
+private ["_muzzles","_eglArray","_magArray","_weapon"];
```
Comment (BoGuu) [inline Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf]: Lets move these down to their respective variables

File: Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf [hunk 2]
```diff
@@ -8,11 +8,20 @@
...
 if ((uiNamespace getVariable ["Weapon_Magazine",0]) isEqualTo 0) then {
-    private ["_weapon"];
     _weapon = lbData[38403,lbCurSel (38403)];
-    _weapon = FETCH_CONFIG2(getArray,"CfgWeapons",_weapon,"magazines");
-    uiNamespace setVariable ["Magazine_Array",_weapon];
+    _magArray = FETCH_CONFIG2(getArray,"CfgWeapons",_weapon,"magazines");
+    _muzzles = FETCH_CONFIG2(getArray,"CfgWeapons",_weapon,"muzzles");
+    if ("EGLM" in _muzzles) then {
```
Comment (BoGuu) [inline Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf]: We can forEach/count over "EGLM" & "GL_3GL_F" to stray from duplicating this code

Comment (Blauschimmel) [inline Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf]: What do you mean by duplicating? It's always either "EGLM" or "GL_3GL_F". The 3gl array includes eglm. Or fo I missunderstand you?

File: Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf (hunk 1 above)
Comment (Blauschimmel) [inline Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf]: You mean below line 12?

File: Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf (hunk 2 above)
Comment (danielstuart14) [inline Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf]: something like that: 
```
disableSerialization;

if ((uiNamespace getVariable ["Weapon_Magazine",0]) isEqualTo 0) then {
    private ["_weapon","_magArray"];
    _weapon = lbData[38403,lbCurSel (38403)];
    _magArray = [];
    {
        if (_x isEqualTo "this") then {
            _magArray append FETCH_CONFIG2(getArray,"CfgWeapons",_weapon,"muzzles");
        } else {
            _magArray append FETCH_CONFIG(getArray,"CfgWeapons",_weapon,_x,"magazines");
        };
    } forEach FETCH_CONFIG2(getArray,"CfgWeapons",_weapon,"muzzles");

    uiNamespace setVariable ["Magazine_Array",_magArray];
    uiNamespace setVariable ["Weapon_Magazine",1];
} else {
    uiNamespace setVariable ["Weapon_Magazine",0];
};
```

**Comment by Blauschimmel** [comment]
@BoGuu You mean this way?

**Comment by BoGuu** [comment]
I haven't looked at the configs yet, and I don't know the format, but you had:

`    if ("EGLM" in _muzzles) then {
        _eglArray = FETCH_CONFIG(getArray,"CfgWeapons",_weapon,"EGLM","magazines");
        _magArray append _eglArray;
    };
    if ("GL_3GL_F" in _muzzles) then {
        _eglArray = FETCH_CONFIG(getArray,"CfgWeapons",_weapon,"GL_3GL_F","magazines");
        _magArray append _eglArray;
    };`

Which can just be:

`{
    if (_x in _muzzles) then {
        _eglArray = FETCH_CONFIG(getArray,"CfgWeapons",_weapon,_x,"magazines");
        _magArray append _eglArray;
    };
    true
} count ["EGLM", "GL_3GL_F"];`

**Comment by danielstuart14** [comment]
just one GL kind will be in the weapon. Or EGLM or 3GL

**Comment by Blauschimmel** [comment]
I'm just not familiar to the count command. I'll check and submit this later. So count is quicker than forEach? And it's better to check if  "EGLM" or "GL_3GL_F" is in muzzles than checking for each muzzle if it's one of these two?

**Comment by Blauschimmel** [comment]
@BoGuu is that true rally necessary?

**Comment by BoGuu** [comment]
> And it's better to check if "EGLM" or "GL_3GL_F" is in muzzles than checking for each muzzle if it's one of these two?

Like I said, I haven't checked the configs, but your original code can be replaced by what I posted. As to whether that is best for what you are trying to do, I don't know `I haven't checked the configs` ;)

But, if "EGLM"/"GL_3GL_F" are the only entries you're checking, then best to check if they are in muzzles (assuming you're right), than looping through all the muzzles, which will inevitably be slower.

@danielstuart14 What do you think? I'll check out the configs today though

**Comment by danielstuart14** [comment]
sounds good.

**Comment by Blauschimmel** [comment]
@BoGuu What do you think?

File: Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf [hunk 3]
```diff
@@ -9,10 +9,15 @@
 disableSerialization;
 
 if ((uiNamespace getVariable ["Weapon_Magazine",0]) isEqualTo 0) then {
-    private ["_weapon"];
+    private ["_weapon","_magArray"];
     _weapon = lbData[38403,lbCurSel (38403)];
-    _weapon = FETCH_CONFIG2(getArray,"CfgWeapons",_weapon,"magazines");
-    uiNamespace setVariable ["Magazine_Array",_weapon];
+    _magArray = FETCH_CONFIG2(getArray,"CfgWeapons",_weapon,"magazines");
```
Comment (BoGuu) [inline Altis_Life.Altis/core/shops/fn_weaponShopMags.sqf]: Can move these privates next to the variables themselves

**Comment by BoGuu** [comment]
Sorry for the late reply, see the comment I just made, and if all is working, LGTM

**Comment by BoGuu** [comment]
All working well @Blauschimmel ?

**Comment by coresync2k** [comment]
tested it, working well

**Comment by Blauschimmel** [comment]
@BoGuu all working fine! I'll do another specificcheck in the Client log when I'm back home but I'm already using it since I posted it and never discovered any issues!

**Comment by Blauschimmel** [comment]
Nothing shown in any log and no issues at all. Tested with all kinds of GL ammo.

**Pull request merged**